imagick/resize.go

```diff
@@ -0,0 +1,141 @@
+package imagick
+
+import (
+	"github.com/journeymidnight/pipa/helper"
+	"github.com/journeymidnight/pipa/library"
+	"gopkg.in/gographics/imagick.v3/imagick"
+	"math"
+)
+
+const Method = imagick.FILTER_LANCZOS
+
+type Resize struct {
+	Width      int
+	Height     int
+	Zoom       float64
+	Force      bool
+	Crop       bool
+	Pad        bool
+	Limit      bool
+	Background string
+}
+
+func (img *ImageWand) resize(o *Resize) (err error) {
+	originWidth := int(img.MagickWand.GetImageWidth())
+	originHeight := int(img.MagickWand.GetImageHeight())
+
+	// image calculations
+	factor := imageCalculations(o, originWidth, originHeight)
+	helper.Log.Info("resize factor: ", factor)
+
+	if o.Limit && !o.Force {
+		if originWidth < o.Width && originHeight < o.Height {
+			factor = 1.0
+			o.Width = originWidth
+			o.Height = originHeight
+		}
+	}
+
+	if o.Zoom != library.Zoom {
```

Comment:
I suggest using switch after abstracting the method name, which is logically a choice